src/frontend/mod.rs

```diff
@@ -1,5 +1,7 @@
 pub mod file_chooser;
 pub mod main_window;
+pub mod rectangle;
```

Comment:
What do you think about creating a directory like `widgets` where all widgets will be stored like `rectangle.rs`, `line.rs`, `arrow.rs` etc..
Then we could also add some abstractions to `widgets/mod.rs` like

```rust
#[derive(Debug, Clone)]
pub struct Point {
    x: f64,
    y: f64,
}
```

and those can be used in `rectangle.rs` as well:

```rust
#[derive(Clone)]
pub struct Rectangle {
    pub top_left: Point,
    pub bottom_right: Point,
}
```

But this can be done in another PR if you agree.

Reply:
Sure, that addition will be included in the upcoming pull request. However, I can implement it now if you prefer. But note that 'Rectangle' isn't a widget; rather, it's simply a struct that represents a rectangle without any inherent drawing capabilities. In any case, I believe we should move away from the term 'widget' and instead use 'drawable,' which aligns with how we reference objects that can be drawn in the codebase.

Reply:
> Sure, that addition will be included in the upcoming pull request.

nice!

> However, I can implement it now if you prefer. 

yes please! :)

> But note that 'Rectangle' isn't a widget; rather, it's simply a struct that represents a rectangle without any inherent drawing capabilities. In any case, I believe we should move away from the term 'widget' and instead use 'drawable,' which aligns with how we reference objects that can be drawn in the codebase.

okeydokey